### PR TITLE
Aadd tokio tests

### PR DESCRIPTION
**Pull request by jettify**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,39 @@
...
+        mkdir ~/rust-installer
+        curl -sL https://static.rust-lang.org/rustup.sh -o ~/rust-installer/rustup.sh
+        sh ~/rust-installer/rustup.sh --prefix=~/rust --spec=$RUST_VERSION -y --disable-sudo
+        export PATH="$HOME/rust/bin:$PATH"
+        export PYTHON_LIB=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))")
+        find $PYTHON_LIB
+        export LIBRARY_PATH="$LIBRARY_PATH:$PYTHON_LIB"
+        export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$PYTHON_LIB:$HOME/rust/lib"
+        rustc -V
+        if [ "$TOKIO" = "y" ]; then
```
Comment (kxepal) [inline .travis.yml]: Does it worth to install rust all the time when TOKIO is `n`? Seems like a waste of a valuable build time.

File: .travis.yml [hunk 2]
```diff
@@ -1,21 +1,39 @@
...
+  allow_failures:
+    - env: TOKIO="y"
 
 install:
     - pip install --upgrade setuptools
     - pip install codecov
     - pip install -Ur requirements-dev.txt
     - python setup.py develop
+    - |
+        export RUST_VERSION=1.17.0
```
Comment (kxepal) [inline .travis.yml]: May be move rust version to matrix?

Comment (jettify) [inline .travis.yml]: sure

File: .travis.yml (hunk 1 above)
Comment (jettify) [inline .travis.yml]: fixed

**Comment by jettify** [comment]
@kxepal thanks for review, merging for now, will fix other concerns if any in next PRs

**Pull request merged**